src/tablestore.rs

```diff
@@ -53,41 +54,52 @@ impl ReadOnlyBlob for ReadOnlyObject {
     }
 
     async fn read_range(&self, range: Range<usize>) -> Result<Bytes, SlateDBError> {
-        self.object_store
-            .get_range(&self.path, range)
+        let opts = CacheableGetOptions {
+            range: Some(GetRange::Bounded(range)),
+            ..Default::default()
+        };
+        let result = self
+            .object_store
+            .get_opts(&self.path, opts)
             .await
-            .map_err(SlateDBError::ObjectStoreError)
+            .map_err(SlateDBError::ObjectStoreError)?;
+        result.bytes().await.map_err(SlateDBError::ObjectStoreError)
     }
 
     async fn read(&self) -> Result<Bytes, SlateDBError> {
-        let file = self.object_store.get(&self.path).await?;
+        let opts = CacheableGetOptions {
+            range: None,
+            ..Default::default()
+        };
+        let file = self.object_store.get_opts(&self.path, opts).await?;
         file.bytes().await.map_err(SlateDBError::ObjectStoreError)
     }
 }
 
 impl TableStore {
     #[allow(dead_code)]
     pub fn new(
-        object_store: Arc<dyn ObjectStore>,
+        object_store: impl Into<CacheableObjectStore>,
         sst_format: SsTableFormat,
         root_path: Path,
     ) -> Self {
         Self::new_with_fp_registry(
-            object_store,
+            object_store.into(),
             sst_format,
             root_path,
             Arc::new(FailPointRegistry::new()),
         )
     }
 
     pub fn new_with_fp_registry(
-        object_store: Arc<dyn ObjectStore>,
+        cacheable_object_store: impl Into<CacheableObjectStore>,
```

Comment:
Should `wal` ssts be cached locally? They are either in ImmutableWal or are part of the memtable, so they are in memory. Is there benefit in writing `wal` to disk?

Reply:
thank you for the information, i think the local cache on WAL could be disabled on the fact that `ImmutableWal` had already been placed in the memory.

i'll change it ensuring the WAL does not use the local cache at all.

Reply:
oh, i found the current WAL store is not wrapped with `CacheableObjectStore`, so i think it's not to expected to cache the WAL files in this pr.